### PR TITLE
estimate tails

### PR DESCRIPTION
**Pull request by 0x00b1**

### CODE REVIEW

File: neuralcompression/functional/_estimate_tails.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import torch
+
+
+def estimate_tails(
+    cdf: typing.Callable[[torch.Tensor], torch.Tensor],
+    target: float,
+    shape: int,
+    device: typing.Optional[typing.Union[int, str, torch.device]] = None,
+    dtype: torch.dtype = torch.float32,
+) -> torch.Tensor:
```
Comment (mmuckley) [inline neuralcompression/functional/_estimate_tails.py:12]: Primary ask is a docstring here describing what the function is doing mathematically - even if it's for internal use.

**Review by mmuckley — CHANGES_REQUESTED** [review]

**Comment by 0x00b1** [comment]
@mmuckley part of the complement of useful, general-purpose ops from Ballé (2018). I'll make a follow-up PR for boundary ops that uses this implementation.

**Comment by mmuckley** [comment]
None of the major powers are watching this repository (yet) so I think the war will hold off for now.

**Comment by mmuckley** [comment]
For this one I'm just waiting on a unit test.

**Review by mmuckley — APPROVED** [review]
LGTM.

File: neuralcompression/functional/_estimate_tails.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
+import typing
```
Comment (mmuckley) [inline neuralcompression/functional/_estimate_tails.py:1]: Please add a copyright header.
```
"""
Copyright (c) Facebook, Inc. and its affiliates.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
"""
```

File: tests/functional/test_estimate_tails.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import math
```
Comment (mmuckley) [inline tests/functional/test_estimate_tails.py:1]: Please add a copyright header.
```
"""
Copyright (c) Facebook, Inc. and its affiliates.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
"""
```

**Review by mmuckley — CHANGES_REQUESTED** [review]

**Review by mmuckley — APPROVED** [review]
LGTM.

**Pull request merged**